ecosystem/typescript/sdk/SPEC.md

```diff
@@ -0,0 +1,220 @@
+  - [5. Coin Management](#5-Coin-Management)
+- Maintenance Requirements
+  - [6. Source Control](#6-source-control)
+  - [7. Releases & Versioning](#7-releases--versioning)
+  - [8. CI Server](#8-ci-server)
+- Additional Content Requirements
+  - [9. Documentation](#9-documentation)
+- Dependencies & Infrastructure Requirements
+  - [10. Testing](#10-testing)
+  - [11. Dependencies](#11-dependencies)
+  - [12. HTTP Client](#12-http-client)
+  - [13. Logging](#13-reporting)
+  - [14. Reporting](#14-reporting)
+- Initialization & Interaction Requirements
+  - [15. Initialization](#15-initalization)
+  - [16. Error Handling](#16-error-handling)
+
+## Base Requirements
+
+### 1. BCS Encoding & Decoding
+
+- [ ] **1.1** The SDK **must** support BCS encoding & decoding for.
+  - [ ] **1.1.1** Unsigned integers u8, u32, u64, u128, u256.
+  - [ ] **1.1.2** Optional arguments.
+  - [ ] **1.1.3** Tuples.
+  - [ ] **1.1.4** Enums.
+  - [ ] **1.1.5** UTF-8 Strings as vector of u8.
+  - [ ] **1.1.6** Structs.
+  - [ ] **1.1.7** Vectors using uleb128.
+  - [ ] **1.1.8** Addresses as a fixed 32 byte type.
+  - [ ] **1.1.9** Generics.
+  - [ ] **1.1.10** bool.
+  - [ ] **1.1.11** Object<T> encodings as an address.
+- [ ] **1.2** The encoding & decoding details **should** be hidden from the end user and done by the SDK.
+
+### 2. API servers
+
+- [ ] **2.1** The SDK **must** interact directly with the Aptos REST API.
+  - [ ] **2.1.1** The SDK **must** support pagination with optional `start` and `limit` parameters to the REST API request.
+  - [ ] **2.1.2** The SDK **should** implement a client that adheres to the OpenAPI spec https://fullnode.mainnet.aptoslabs.com/v1/spec#/.
+- [ ] **2.2** The SDK **should** interact directly with the Aptos Indexer API.
+  - [ ] **2.2.1** The SDK **must** support pagination with optional `offset` and `limit` parameters to Indexer API request.
+  - [ ] **2.2.2** The SDK **must** validate the account address when interacting with the Indexer API.
+    - [ ] **2.2.2.1** The SDK **must** validate an account address is a 64 character hex string with a leading `0x`.
+  - [ ] **2.2.3** The SDK **must** handle premade GraphQL queries https://cloud.hasura.io/public/graphiql?endpoint=https://indexer.mainnet.aptoslabs.com/v1/graphql.
+  - [ ] **2.2.3** The SDK **should** allow the user to pass in a custom GraphQL query.
+- [ ] **2.3** The SDK **should** support interactions with Aptos devnet, testnet and mainnet networks.
+- [ ] **2.4** The SDK **should** support interactions with a custom URL.
+
+### 3. Account Management
+
+- [ ] **3.1** The SDK **must** support account creation.
+- [ ] **3.2** The SDK **must** support account APT coin balance read.
+- [ ] **3.3** The SDK **must** support key management for Ed25519 keys.
+  - [ ] **3.3.1** The SDK **must** support Key generation
+  - [ ] **3.3.2** The SDK **must** support Key loading from a file
+  - [ ] **3.3.3** The SDK **must** support Key loading from an input
```

Comment:
Maybe this should just be loaded from a string or byte array? Input is very much a user driven operation as there are dozens of ways to define input.